packages/api/blocks-api/src/blocks/block.ts

```diff
@@ -67,8 +68,8 @@ export abstract class BlockData implements BlockDataInterface {
         return this as any; // MUST NOT transform it's child blocks (these handle transforming themselves)
     }
 
-    indexData(): BlockIndexData {
-        return {};
+    indexData(): BlockIndexDataArray {
+        return [];
```

Comment:
Why do you change this to an array?

Reply:
Because a block could theoretically have multiple dependencies

```
indexData(): BlockIndexDataArray {
    return [
        {
            targetIdentifier: DAM_FILE_BLOCK_INDEX_IDENTIFIER,
            id: this.damFileId,
        },
        {
            targetIdentifier: NEWS_BLOCK_INDEX_IDENTIFIER,
            id: this.newsId,
        },
    ];
}
```

The previous schema was this:

```
indexData(): BlockIndexData {
        return {
            damFileIds: this.damFileId ? [this.damFileId] : [],
        };
    }
```

That doesn't work anymore because we want to be flexible enough to have any kind of dependency.

Reply:
Okay, when thinking about it a schema like this would also be possible:

```
indexData() {
    return {
        [DAM_FILE_BLOCK_INDEX_IDENTIFIER]: this.damFileId,
        [NEWS_BLOCK_INDEX_IDENTIFIER]: this.newsId,
    };
}
```

Thats maybe preferable?

Reply:
> Thats maybe preferable?

no, because there could be multiple files

Reply:
My thoughts on the original BlockIndexData:

- I used a non-flexible interface containing damFileIds to have good typings
- that was currently for damFile dependencies only
- with any other data that can be indexed in future (not only dependencies)
- this can also be done app specific using module augmentation

This PR turns it a bit around:
1. there can now be dependencies to any entity (implemented also for news)
2. indexData is now for dependencies only

- (1.) is a clear win and will have many use cases.
- (2.) is probably loss, however I don't really have many use cases in mind
    - index out of which a fulltext search index is generated
    - index where we can find usages of specific blocks

I see two possibilities:

1. we decide to support dependencies only: the whole indexData feature needs to be renamed to dependencies or so
2. we decide we need both and change the BlockIndexData to something like
```
export interface BlockIndexData {
    dependencies: Array<{
        targetIdentifier: string,
        id: string,
    }>
}
```